doc/guides/device_mgmt/smp_groups/smp_group_8.rst

```diff
@@ -216,3 +220,209 @@ where:
     | "rc"                  | :ref:`mcumgr_smp_protocol_status_codes`           |
     +-----------------------+---------------------------------------------------+
+
+File status
+***********
+
+Command allows to retrieve status of an existing file from specified path
+of a target device.
+
+File status request
+===================
+
+File status request header:
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``0``  | ``8``        |  ``1``         |
+    +--------+--------------+----------------+
+
+CBOR data of request:
+
+.. code-block:: none
+
+    {
+        (str)"name" : (str)
+    }
+
+where:
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "name"                | absolute path to a file                           |
+    +-----------------------+---------------------------------------------------+
+
+File status response
+====================
+
+File status response header:
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``1``  | ``8``        |  ``1``         |
+    +--------+--------------+----------------+
+
+CBOR data of successful response:
+
+.. code-block:: none
+
+    {
+        (str)"len"      : (uint)
+        (str)"rc"       : (int)
+    }
+
+In case of error the CBOR data takes form:
+
+.. code-block:: none
+
+    {
+        (str)"rc"           : (int)
+    }
+
+where:
+
+.. table::
+    :align: center
+
+    +-----------------------+---------------------------------------------------+
+    | "len"                 | length of file (in bytes)                         |
+    +-----------------------+---------------------------------------------------+
+    | "rc"                  | :ref:`mcumgr_smp_protocol_status_codes`           |
+    +-----------------------+---------------------------------------------------+
+
+In case when "rc" is not 0, success, the other fields will not appear.
+
+fuck
+
+File hash/checksum
+******************
+
+Command allows to generate a hash/checksum of an existing file at a specified
+path on a target device. Note that kernel heap memory is required for buffers to
+be allocated for this to function, and large stack memory buffers are required
+for generation of the output hash/checksum.
+
+File hash/checksum request
+==========================
+
+File hash/checksum request header:
+
+.. table::
+    :align: center
+
+    +--------+--------------+----------------+
+    | ``OP`` | ``Group ID`` | ``Command ID`` |
+    +========+==============+================+
+    | ``0``  | ``8``        |  ``2``         |
+    +--------+--------------+----------------+
+
+CBOR data of request:
+
+.. code-block:: none
+
+    {
+        (str)"type"     : (str)
```

Comment:
Make type optional. The type is returned in response anyway and very often users will only rely on whatever default type would be set.

Reply:
Done, made type selection automatic however.